ext/standard/math.c

```diff
@@ -1156,7 +1157,17 @@ PHP_FUNCTION(number_format)
 		thousand_sep_len = 1;
 	}
 
-	RETURN_STR(_php_math_number_format_ex(num, (int)dec, dec_point, dec_point_len, thousand_sep, thousand_sep_len));
+#if SIZEOF_ZEND_LONG > SIZEOF_INT
```

Comment:
You can use ZEND_LONG_INT_OVFL & ZEND_LONG_INT_UDFL which will allow you to avoid this #if and make the code a bit cleaner. These macros will resolve to false is zend_long and int are the same size.
There's also ZEND_LONG_EXCEEDS_INT, but that doesn't apply here because you still want to set dec_int to a sane value.